solc_select/solc_select.py

```diff
@@ -1,30 +1,139 @@
+        )
+        return result.returncode == 0
+    except (FileNotFoundError, OSError):
+        return False
+
+
+def get_emulation_prefix() -> list:
+    """Get the command prefix for emulation if needed."""
+    if get_arch() != "arm64":
+        return []
+
+    # On macOS, use arch command for Rosetta 2
+    if sys.platform == "darwin" and check_emulation_available():
+        return ["arch", "-x86_64"]
+
+    # On Linux, use qemu
+    if sys.platform.startswith("linux") and check_emulation_available():
+        return ["qemu-x86_64"]
+
+    return []
+
+
+def warn_about_arm64(force: bool = False) -> None:
+    """Warn ARM64 users about compatibility and suggest solutions."""
+    if get_arch() != "arm64":
+        return
+
+    # Check if we've already warned
+    warning_file = SOLC_SELECT_DIR.joinpath(".arm64_warning_shown")
+    if not force and warning_file.exists():
+        return
+
+    print("\n⚠️  WARNING: ARM64 Architecture Detected", file=sys.stderr)
+    print("=" * 50, file=sys.stderr)
+
+    if check_emulation_available():
+        if sys.platform == "darwin":
+            print("✓ Rosetta 2 detected - will use emulation for x86 binaries", file=sys.stderr)
+        else:
+            print("✓ qemu-x86_64 detected - will use emulation for x86 binaries", file=sys.stderr)
+        print("  Note: Performance will be slower than native execution", file=sys.stderr)
+    else:
+        if sys.platform == "darwin":
+            print(
+                "✗ solc binaries are x86_64 only, and Rosetta 2 is not available", file=sys.stderr
+            )
+        else:
+            print("✗ solc binaries are x86_64 only, and qemu is not installed", file=sys.stderr)
+        print("\nTo use solc-select on ARM64, you can:", file=sys.stderr)
+        print("  1. Install qemu for x86_64 emulation:", file=sys.stderr)
+        if sys.platform.startswith("linux"):
+            print("     sudo apt-get install qemu-user-static  # Debian/Ubuntu", file=sys.stderr)
+            print("     sudo dnf install qemu-user-static      # Fedora", file=sys.stderr)
+            print("     sudo pacman -S qemu-user-static        # Arch", file=sys.stderr)
+        elif sys.platform == "darwin":
+            print("     Use Rosetta 2 (installed automatically on Apple Silicon)", file=sys.stderr)
+        print("  2. Use an x86_64 Docker container", file=sys.stderr)
+        print("  3. Use a cloud-based development environment", file=sys.stderr)
+    print("=" * 50, file=sys.stderr)
+    print(file=sys.stderr)
+
+    # Mark that we've shown the warning
+    with contextlib.suppress(OSError):
+        warning_file.touch()
+
+
+def validate_url_scheme(url: str) -> None:
+    """Validate that URL uses a safe scheme (http or https only)."""
+    parsed = urlparse(url)
+    if parsed.scheme not in ("http", "https"):
+        raise ValueError(f"Unsafe URL scheme '{parsed.scheme}' in URL: {url}")
+
```

Comment:
URLs are always constructed internally by `get_url` so they'll always be https, this check would never fail.

Reply:
Spot on! Since `get_url` always produces HTTPS URLs internally, this validation was indeed unnecessary. Removed the `validate_url_scheme` function and all its calls.